Brokerages/Exante/ExanteBrokerage.cs

```diff
@@ -0,0 +1,420 @@
+                        $"ExanteBrokerage.ConvertOrderDuration: Unsupported order duration: {order.TimeInForce}");
+            }
+
+            var quantity = Math.Abs(order.Quantity);
+
+            WebCallResult<IEnumerable<ExanteOrder>> orderPlacement;
+            switch (order.Type)
+            {
+                case OrderType.Market:
+                    orderPlacement = _client.PlaceOrder(
+                        _accountId,
+                        _symbolMapper.GetBrokerageSymbol(order.Symbol),
+                        ExanteOrderType.Market,
+                        orderSide,
+                        quantity,
+                        orderDuration,
+                        gttExpiration: goodTilDateTimeInForceExpiration
+                    );
+                    break;
+
+                case OrderType.Limit:
+                    var limitOrder = (LimitOrder) order;
+                    orderPlacement = _client.PlaceOrder(
+                        _accountId,
+                        _symbolMapper.GetBrokerageSymbol(order.Symbol),
+                        ExanteOrderType.Limit,
+                        orderSide,
+                        quantity,
+                        orderDuration,
+                        limitPrice: limitOrder.LimitPrice,
+                        gttExpiration: goodTilDateTimeInForceExpiration
+                    );
+                    break;
+
+                case OrderType.StopMarket:
+                    var stopMarketOrder = (StopMarketOrder) order;
+                    orderPlacement = _client.PlaceOrder(
+                        _accountId,
+                        _symbolMapper.GetBrokerageSymbol(order.Symbol),
+                        ExanteOrderType.Stop,
+                        orderSide,
+                        quantity,
+                        orderDuration,
+                        stopPrice: stopMarketOrder.StopPrice,
+                        gttExpiration: goodTilDateTimeInForceExpiration
+                    );
+                    break;
+
+                case OrderType.StopLimit:
+                    var stopLimitOrder = (StopLimitOrder) order;
+                    orderPlacement = _client.PlaceOrder(
+                        _accountId,
+                        _symbolMapper.GetBrokerageSymbol(order.Symbol),
+                        ExanteOrderType.Stop,
+                        orderSide,
+                        quantity,
+                        orderDuration,
+                        limitPrice: stopLimitOrder.LimitPrice,
+                        stopPrice: stopLimitOrder.StopPrice,
+                        gttExpiration: goodTilDateTimeInForceExpiration
+                    );
+                    break;
+
+                default:
+                    throw new NotSupportedException(
+                        $"ExanteBrokerage.ConvertOrderType: Unsupported order type: {order.Type}");
+            }
+
+            foreach (var o in orderPlacement.Data)
```

Comment:
There's a race condition between placing the order and getting the event through the callback `GetOrdersStreamAsync`, suggest using a `BrokerageConcurrentMessageHandler` instance to handle this, we've done this for multiple brokerages can see examples of it

Reply:
Sorry, I don't see a race condition inside my code. Could you clarify? Does it happen inside `OnOrderEvent`?

Reply:
> Sorry, I don't see a race condition inside my code. Please, explain where it comes from. Does it happen inside OnOrderEvent?

Sure, when you place a trade you add it to the `_orderMap` and the `GetOrdersStreamAsync` is reading from `_orderMap`. What if `GetOrdersStreamAsync` is triggered before the order is added to `_orderMap` it won't find it and not trigger the order event

Reply:
> Sure, when you place a trade you add it to the _orderMap and the GetOrdersStreamAsync is reading from _orderMap. What if GetOrdersStreamAsync is triggered before the order is added to _orderMap it won't find it and not trigger the order event

Thanks for the explanation. You're right.